R/help.R

```diff
@@ -185,17 +185,31 @@ help_completion_parameter_handler.python.builtin.object <- function(source) {
 
   if (!py_available())
     return(NULL)
-
+  
+  # handle Python functions as-is
+  if (inherits(source, "python.builtin.object"))
+    target <- source
+  
   # split into topic and source
-  components <- source_components(source)
-  if (is.null(components))
-    return(NULL)
-  topic <- components$topic
-  source <- components$source
-
-  # get the function
-  target <- help_get_attribute(source, topic)
-  if (!is.null(target) & py_is_callable(target)) {
+  if (is.character(source)) {
+    components <- source_components(source)
+    if (is.null(components))
+      return(NULL)
+    topic <- components$topic
+    source <- components$source
+    
+    # get the function
+    target <- help_get_attribute(source, topic)
+  }
+  
+  # if this is a class object, use the __init__ method
+  if (inherits(target, "python.builtin.type")) {
+    init <- py_get_attr(target, "__init__", silent = TRUE)
+    if (!is.null(init))
+      target <- init
```

Comment:
We had something similar to this for the wrapper generation functions. But one potential issue could be, e.g. this [Estimator class](https://github.com/tensorflow/tensorflow/blob/e3d63c81f15f962c0841b3a6f97b7df93b0343cf/tensorflow/python/estimator/estimator.py#L70) that we are familiar with have docstrings in both `__init__` and the class. This is very common in other libraries too. I think we can:
1. concatenate both docs for the `description` field that we are extracting in some way
2. extract argument descriptions from `__init__` for classes.

Reply:
The reason for (1) is that if we only extract doc string from `__init__`, this will be the only thing showing up "Constructs an `Estimator` instance.". However, users won't know what an `Estimator` is, which is where they need to look up the doc-string for `Estimator` class in addition to `__init__`.
 
I saw in RStudio there's a "press F1 for additional help" when the help pops up. Instead of concatenate as I suggested in (1), Is there a way for us to only show the class descriptions when pressing F1? Sometimes the descriptions are quite long.